src/components/composer/content_editable_helper.ts

```diff
@@ -48,8 +52,16 @@ export class ContentEditableHelper {
       }
       let startNode = this.findChildAtCharacterIndex(start);
       let endNode = this.findChildAtCharacterIndex(end);
-      range.setStart(startNode.node, startNode.offset);
-      range.setEnd(endNode.node, endNode.offset);
+
+      // setEnd (setStart) will result in a collapsed range if the end point is before the start point
+      // https://developer.mozilla.org/en-US/docs/Web/API/Range/setEnd
+      if (start <= end) {
+        range.setStart(startNode.node, startNode.offset);
+        range.setEnd(endNode.node, endNode.offset);
+      } else {
+        range.setStart(endNode.node, endNode.offset);
+        range.setEnd(startNode.node, startNode.offset);
```

Comment:
@rrahir Do you recall why you changed `extend` to `setEnd` [here](https://github.com/odoo/o-spreadsheet/pull/5012/files#diff-0c54baeea98218a3cf48db2bfac7a4cb31c4a8ee0406e314c08f996bf7524e73) ? I'm guessing you had a good reason

Reply:
I really dont. If I had to guess, because I could'nt make it work as I wanted in FF for some reason.. i'll have a look